test/lazy/_lazy_tensor_test_case.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import gpytorch
+import torch
+import os
+import random
+from abc import abstractmethod
+from gpytorch.utils import approx_equal
+
+
+class LazyTensorTestCase(object):
+    should_test_sample = False
+
+    @abstractmethod
+    def create_lazy_tensor(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def evaluate_lazy_tensor(self):
+        raise NotImplementedError()
+
+    def setUp(self):
+        if hasattr(self.__class__, "seed"):
+            seed = self.__class__.seed
+            if os.getenv("UNLOCK_SEED") is None or os.getenv("UNLOCK_SEED").lower() == "false":
+                self.rng_state = torch.get_rng_state()
+                torch.manual_seed(seed)
+                if torch.cuda.is_available():
+                    torch.cuda.manual_seed_all(seed)
+                random.seed(seed)
+
+    def tearDown(self):
+        if hasattr(self, "rng_state"):
+            torch.set_rng_state(self.rng_state)
+
+    def test_matmul_vec(self):
+        lazy_tensor = self.create_lazy_tensor()
+        lazy_tensor_copy = lazy_tensor.clone()
+        evaluated = self.evaluate_lazy_tensor(lazy_tensor_copy)
+
+        test_vector = torch.randn(lazy_tensor.size(-1))
+        res = lazy_tensor.matmul(test_vector)
+        actual = evaluated.matmul(test_vector)
+        assert(approx_equal(res, actual))
```

Comment:
lets make this a `self.assertTrue(approx_equal(res, actual))` (and similarly in other places)